bindings/wasm/src/account/types/key_location.rs

```diff
@@ -4,56 +4,73 @@
 use identity::account_storage::KeyLocation;
 use wasm_bindgen::prelude::*;
 
-use crate::account::types::WasmGeneration;
-use crate::did::WasmMethodType;
+use crate::crypto::WasmKeyType;
+use crate::did::WasmVerificationMethod;
 use crate::error::Result;
 use crate::error::WasmResult;
 
+
+/// The storage location of a verification method key.
+///
+/// A key is uniquely identified by the fragment and a hash of its public key.
+/// Importantly, the fragment alone is insufficient to represent the storage location.
+/// For example, when rotating a key, there will be two keys in storage for the
+/// same identity with the same fragment. The `key_hash` disambiguates the keys in
+/// situations like these.
+///
+/// The string representation of that location can be obtained via `canonicalRepr`.
+#[derive(Debug)]
 #[wasm_bindgen(js_name = KeyLocation, inspectable)]
 pub struct WasmKeyLocation(pub(crate) KeyLocation);
 
 #[wasm_bindgen(js_class = KeyLocation)]
 impl WasmKeyLocation {
+  /// Create a location from a `KeyType`, the fragment of a verification method
+  /// and the bytes of a public key.
   #[wasm_bindgen(constructor)]
-  pub fn new(method: &WasmMethodType, fragment: String, generation: &WasmGeneration) -> WasmKeyLocation {
-    WasmKeyLocation(KeyLocation::new(method.0, fragment, generation.0))
-  }
-
-  /// Returns a copy of the method type of the key location.
-  #[wasm_bindgen]
-  pub fn method(&self) -> WasmMethodType {
-    self.0.method().into()
+  pub fn new(key_type: WasmKeyType, fragment: String, public_key: Vec<u8>) -> WasmKeyLocation {
```

Comment:
Use camelCase for Wasm parameters.

```suggestion
  #[allow(non_snake_case)]
  pub fn new(keyType: WasmKeyType, fragment: String, publicKey: Vec<u8>) -> WasmKeyLocation {
```